### PR TITLE
Feed Example: Move display of example from separate page into standard APG example page

### PR DESCRIPTION
**Pull request by ariellalgilmore**
Closes: #2747 

This moves the feed example inside the page using an `iframe`
Screenshot 2023-08-28 at 1 39 19 PM

- add an iframe linking to the `frame-display.html`
- deleted unused `imgs` folder
- updated tests

worked on with @andreancardona 🎉 !

#### Preview link

[Infinite Scrolling Feed Example | APG | WAI | W3C](https://deploy-preview-248--aria-practices.netlify.app/aria/apg/patterns/feed/examples/feed/)

#### Review checklist

*Reviewers:* To learn what needs to be covered by each review, Follow the link for the type of review to which you are assigned.

* [ ] [Editorial review](https://github.com/w3c/aria-practices/wiki/Pull-Request-Review-Process#editorial_review) by @mcking65 (plus keyboard navigation)
* [ ] [Functional review](https://github.com/w3c/aria-practices/wiki/Pull-Request-Review-Process#functional_review) by @jongund  
* [x] [Visual design review](https://github.com/w3c/aria-practices/wiki/Pull-Request-Review-Process#design_review) by @a11ydoer 
* [x] [Accessibility review](https://github.com/w3c/aria-practices/wiki/Pull-Request-Review-Process#accessibility_review)  @a11ydoer 
* [x] [Code review](https://github.com/w3c/aria-practices/wiki/Pull-Request-Review-Process#code_review) by @howard-e 
* [x] [Test review](https://github.com/w3c/aria-practices/wiki/Pull-Request-Review-Process#test_review) by @howard-e 

___
[WAI Preview Link](https://deploy-preview-248--aria-practices.netlify.app/ARIA/apg) _(Last built on Tue, 21 May 2024 13:38:10 GMT)._

### CODE REVIEW

**Comment by mcking65** [comment]
Hi @ariellalgilmore, it seems atypical that a feed would be in a dialog and more likely that it would be integrated into the body of a page. I am curious what inspired the dialog approach? We can discuss in tomorrow's meeting ... hopefully.

**Comment by ariellalgilmore** [comment]
> Hi @ariellalgilmore, it seems atypical that a feed would be in a dialog and more likely that it would be integrated into the body of a page. I am curious what inspired the dialog approach? We can discuss in tomorrow's meeting ... hopefully.

Hi @mcking65, the dialog approach was inspired because if we integrated the feed example directly in the page I wasn't sure how users would ever be able to reach the rest of the content within the page since the feed example would continue to add more restaurant recommendations as a user scrolls.

File: content/patterns/feed/examples/feed.html
Comment (a11ydoer) [inline content/patterns/feed/examples/feed.html:1]: Thanks for PR, @ariellalgilmore and @andreancardona!

To me, the iframe is broken and it has "Page not found" error,  in Feed example page. I think it is due to the incorrect path to "feed-display.html".

Can you check the path to the feed-display.html in the source`src="./feed-display.html"` correct? I think feed-display.html file path should be under "example" folder, not under second "feed" folder.

**Review by a11ydoer — CHANGES_REQUESTED** [review]
Thanks for PR, @ariellalgilmore and @andreancardona!

To me, the iframe is broken and it has "Page not found" error,  in Feed example page. I think it is due to the incorrect path to "feed-display.html".

Can you check the path to the feed-display.html in the source`src="./feed-display.html"` correct? I think feed-display.html file path should be under "example" folder, not under second "feed" folder.
page not found error - feed example

**Comment by ariellalgilmore** [comment]
Thanks @a11ydoer! Weird the deploy preview is acting different then my local and the testing environment it seems. Deploy preview should work now, but the tests are breaking 😞

**Comment by a11ydoer** [comment]
> Thanks @a11ydoer! Weird the deploy preview is acting different then my local and the testing environment it seems. Deploy preview works now, but the tests are breaking 😞

@ariellalgilmore Thanks for fixing the path. It looks good. Don't worry about regression test. It is failing because there is no "feed" role information in the testing file.

File: content/patterns/feed/examples/feed.html [hunk 1]
```diff
@@ -39,11 +39,29 @@ <h2>About This Example</h2>
...
+        </fieldset>
+      </section>
+
       <section>
         <h2 id="ex_label">Example</h2>
         <div role="separator" id="ex_start_sep" aria-labelledby="ex_start_sep ex_label" aria-label="Start of"></div>
         <div id="ex1">
-          <p>The example feed experience is presented on a separate <a href="feed-display.html">feed display page.</a></p>
+          <p>The example feed experience below is presented in an iframe in order not to obstruct from the rest of the content of the page. </p>
+          <iframe id="feed_frame" src="./feed-display.html" height="500" title="Feed example" style="width: 100%"></iframe>
```
Comment (ariellalgilmore) [inline content/patterns/feed/examples/feed.html:64]: Hi @jugglinmike, the deploy preview will only work if I change the following:
```suggestion
          <iframe id="feed_frame" src="../feed-display.html" height="500" title="Feed example" style="width: 100%"></iframe>
```

but keeping it as `src="./feed-display.html"` gets the tests to correctly run and has it successfully work when viewing locally.

Do you have any ideas on how to go about fixing this?

Comment (jugglinmike) [reply content/patterns/feed/examples/feed.html:64]: Hey there, Ariella! Although we were suspicious of the automated testing infrastructure during today's call, this appears to actually be a snag in the build process. I've diagnosed the problem and [proposed a solution](https://github.com/w3c/wai-aria-practices/pull/262); if that's accepted, then the preview should start to work using `./feed-display.html` as you expected (though you'll likely need to push another commit to trigger the system to regenerate).

Comment (ariellalgilmore) [reply content/patterns/feed/examples/feed.html:64]: Thanks @jugglinmike!

**Comment by mcking65** [comment]
I have one other question ... is the open in codepen functionality going to work with an iframe? We don't have that on this example yet.

This might be a ridiculous question ... or maybe not ... what if we eliminated the feed-display.html file altogether and integrated the example directly into the page? Would it be possible to get the same presentation? Would that create problems that are harder or easier to solve?

**Comment by mcking65** [comment]
@ariellalgilmore

I made some editorial changes to the example usage option description.

I also removed the fieldset. Since there is only one option, the grouping provided by the fieldset is superfluous. This makes the presentation of the usage option similar to [Example Disclosure Navigation Menu with Top-Level Links](https://www.w3.org/WAI/ARIA/apg/patterns/disclosure/examples/disclosure-navigation-hybrid/), which is another example that has only one usage control.

In addition, I moved the label in the reading order to be before the select. I have no idea if repositioning the label in the DOM messed up anything visually. If so, please fix the visual presentation.

File: content/patterns/feed/examples/feed.html (hunk 1 above)
Comment (a11ydoer) [reply content/patterns/feed/examples/feed.html:59]: @ariellalgilmore @jugglinmike @mcking65 

What is the correct example for feed example link for the reivew? 
https://deploy-preview-248--aria-practices.netlify.app/aria/apg/patterns/feed/examples/feed-display does not show iframe feed example. The file path is aria/apg/patterns/feed/examples/feed-display

https://deploy-preview-248--aria-practices.netlify.app/aria/apg/patterns/feed/examples/feed/ still show page not fouud error.

@mcking65 The code pen button would for work for this example.

**Comment by a11ydoer** [comment]
> I have one other question ... is the open in codepen functionality going to work with an iframe? We don't have that on this example yet.
> I think codepen button would work. We can test it out. 

> This might be a ridiculous question ... or maybe not ... what if we eliminated the feed-display.html file altogether and integrated the example directly into the page? Would it be possible to get the same presentation? Would that create problems that are harder or easier to solve?

I thought it was confusing to have pattern/feed and pattern/feed/example/feed-display so I had the same idea at first, @mcking65, just one page for Feed.  

However, for the consistency of the APG directory structure, it is necessary to have the separate feed example under the example folder. 

Currently we have the folder structure of 

- patterns/disclosure/
- - patterns/disclosure/examples/disclosure-faq/
- - patterns/disclosure/examples/disclosure-image-description/

Even though we have only example under the pattern, we still keep the same folder structure. 

- patterns/alert/
-- patterns/alert/examples/alert/

@mcking65 what do you mean by "same presentation"? You might mean the same template view such as content left side and page content list on the right side?

**Comment by mcking65** [comment]
> > I have one other question ... is the open in codepen functionality going to work with an iframe? We don't have that on this example yet.
> > I think codepen button would work. We can test it out.

It seems to me it would be very complicated with an iframe if it is possible at all.

> > This might be a ridiculous question ... or maybe not ... what if we eliminated the feed-display.html file altogether and integrated the example directly into the page? Would it be possible to get the same presentation? Would that create problems that are harder or easier to solve?
> 
> I thought it was confusing to have pattern/feed and pattern/feed/example/feed-display so I had the same idea at first, @mcking65, just one page for Feed.
> 
> However, for the consistency of the APG directory structure, it is necessary to have the separate feed example under the example folder.
>... 
> Even though we have only one example under the pattern, we still keep the same folder structure.

@a11ydoer I didn't state it clearly. I was not suggesting to eliminate the example page. I was asking about eliminating the feed-display.html file.

Currently we have:
* content/patterns/feed/examples/feed-display.html
* content/patterns/feed/examples/feed.html

Perhaps we only need content/patterns/feed/examples/feed.html.

> @mcking65 what do you mean by "same presentation"? You might mean the same template view such as content left side and page content list on the right side?

No, I'm asking if we eliminate the feed-display.html file and the iframe, will the feed itself still have the same presentation as it does in the current iframe?

**Comment by a11ydoer** [comment]
> > > I have one other question ... is the open in codepen functionality going to work with an iframe? We don't have that on this example yet.
> > > I think codepen button would work. We can test it out.
> 
> It seems to me it would be very complicated with an iframe if it is possible at all.
> 
> > > This might be a ridiculous question ... or maybe not ... what if we eliminated the feed-display.html file altogether and integrated the example directly into the page? Would it be possible to get the same presentation? Would that create problems that are harder or easier to solve?
> > 
> > 
> > I thought it was confusing to have pattern/feed and pattern/feed/example/feed-display so I had the same idea at first, @mcking65, just one page for Feed.
> > However, for the consistency of the APG directory structure, it is necessary to have the separate feed example under the example folder.
> > ...
> > Even though we have only one example under the pattern, we still keep the same folder structure.
> 
> @a11ydoer I didn't state it clearly. I was not suggesting to eliminate the example page. I was asking about eliminating the feed-display.html file.
> 
> Currently we have:
> 
> * content/patterns/feed/examples/feed-display.html
> * content/patterns/feed/examples/feed.html
> 
> Perhaps we only need content/patterns/feed/examples/feed.html.
> 
Yes, Let's remove feed-display.html since we have the infinite scroll example as the iframe and it will reduced the confusion about which one is the recent one.

> > @mcking65 what do you mean by "same presentation"? You might mean the same template view such as content left side and page content list on the right side?
> 
> No, I'm asking if we eliminate the feed-display.html file and the iframe, will the feed itself still have the same presentation as it does in the current iframe?

Yes, it does.

**Comment by mcking65** [comment]
@a11ydoer wrote:

> Yes, Let's remove feed-display.html since we have the infinite scroll example as the iframe and it will reduced the confusion about which one is the recent one.

I don't think there could be confusion about what is current. The current architecture is that the feed.html file includes an iframe that shows the example that is encoded in the fee-display.html file. So, there is no redundancy or potential for confusion.

My question is whether we want a two-file architecture or a single-file architecture for the feed example.

As I think about this more, there could be real advantages to a two-file architecture not only for this example but all examples. In #2386, we have a request that all examples have a bare-bones presentation without any of the documentation. I think that is an important request, and I now wonder if having a two-file architecture for all examples that uses iframes might be a good way to address it. This PR is well-positioned as a prototype for how we could address #2386 with a two-file architecture.

Maybe we could change how the codepen feature works so it opens the feed-display.html from feed.html. We could generate some standard APG metadata that governs that behavior for the codepen button.

To create a solution prototype for #2386, we would also need:
* An appropriately named link for opening feed-display.html, e.g., "Open example without documentation in a separate window".
* A back-link in feed-display.html that returns to the feed.html page, e.g. "View example with supporting documentation"
* Possibly a way to suppress some of the site-wide template so that the bare-bones page is as bare-bones as possible but still has sufficient WAI site branding. That would require discussion with the WAI team.

To be clear, these are just ideas. I'm not asking @ariellalgilmore to make any of these changes. I think they deserve further discussion first. I have it on tomorrow's agenda.

@howard-e If you have time to look at how the current codepen functionality works and think how it could work for pages where the example code is in a separate file, that would be helpful. We would want the codepen button in two places: on the feed.html page just as it is for all other example pages, and on the feed-display.html page so that when people open that page, they can still open it in codepen. In both cases, we would want the content going to codepen to be identical.

> > @mcking65 what do you mean by "same presentation"? You might mean the same template view such as content left side and page content list on the right side?
> > 
> > 
> > No, I'm asking if we eliminate the feed-display.html file and the iframe, will the feed itself still have the same presentation as it does in the current iframe?
> 
> Yes, it does.

Do you mean it could if we changed this example to a single-file architecture? I wonder if it would or if there would be styling advantages or disadvantages of a single-file approach.

**Comment by howard-e** [comment]
@ariellalgilmore @mcking65 @a11ydoer @jugglinmike I took a look at the iframe issue following yesterday's APG discussion. It now renders as expected on https://deploy-preview-248--aria-practices.netlify.app/aria/apg/patterns/feed/examples/feed/ for me. Please confirm.

**Comment by howard-e** [comment]
There is a separate build error here to look into as well:

<details>
<summary>Error Message: `Timeout: Focus did not reach item at index 0 of items: #delay-time-select␊Wait timed out after 10163ms`</summary>

```
not ok 223 - feed_feed › content/patterns/feed/examples/feed.html [data-test-id="key-control-end"]: CONTROL+END moves focus out of feed
  ---
    name: AssertionError
    message: Rejected promise returned by test
    values:
      'Rejected promise returned by test. Reason:': |-
        TimeoutError {
          remoteStacktrace: '',
          message: `Timeout: Focus did not reach item at index 0 of items: #delay-time-select␊
          Wait timed out after 10163ms`,
        }
    at: |-
      Wait timed out after 10163ms
      node_modules/selenium-webdriver/lib/webdriver.js:929:17
  ...
```

</details>

**Comment by a11ydoer** [comment]
> Matt_King: To solve 2386, we could write some code that extracts and duplicate that content during the build process

Thanks for the fix! @howard-e 
APG group was also discussing a sort of template change to accomodate above requirment. Any expert opinions?

**Comment by howard-e** [comment]
> APG group was also discussing a sort of template change to accomodate above requirment. Any expert opinions?

@a11ydoer no thoughts on how to get it done right now. Should be possible but will certainly take some investigation. Will share any follow up thoughts on this!

**Comment by ariellalgilmore** [comment]
@mcking65 @jugglinmike thoughts on the new fake footer after the feed? I see the regression test is still failing though

**Comment by howard-e** [comment]
> I see the regression test is still failing though

@ariellalgilmore the failing test is reported as:

<details>
<summary>"feed_feed › content/patterns/feed/examples/feed.html [data-test-id="key-control-end"]: CONTROL+END moves focus out of feed" with the error message, "Timeout: focus did not reach item at index 0 of items: #delay-time-select ..."</summary>

not ok 231 - feed_feed › content/patterns/feed/examples/feed.html [data-test-id="key-control-end"]: CONTROL+END moves focus out of feed
  ---
    name: AssertionError
    message: Rejected promise returned by test
    values:
      'Rejected promise returned by test. Reason:': |-
        TimeoutError {
          remoteStacktrace: '',
          message: `Timeout: Focus did not reach item at index 0 of items: #delay-time-select␊
          Wait timed out after 10189ms`,
        }
    at: |-
      Wait timed out after 10189ms
      node_modules/selenium-webdriver/lib/webdriver.js:929:17
  ...

</details>

Because [`navigateToFeed` now switches the session's context to the `#feed_frame` iframe](https://github.com/w3c/aria-practices/blob/f0aaf774ca8e69f585a8dea3bc8c17a0a601ea52/test/tests/feed_feed.js#L22), the test runner now has to [leave that iframe's context](https://www.selenium.dev/documentation/webdriver/interactions/frames/#leaving-a-frame) to return to the main window's context to "find" `#delay-time-select`. This one failed because it's the only test in the suite that attempts to interact with an element outside of the iframe.

An update to `feed_feed.js` similar to the following snippet should do the trick:

```diff
ariaTest(
  'CONTROL+END moves focus out of feed',
  exampleFile,
  'key-control-end',
  async (t) => {
    await navigateToFeed(t);
    await waitForArticlesToLoad(t);

    let articles = await t.context.queryElements(t, ex.articleSelector);
-   articles[0].sendKeys(Key.chord(Key.CONTROL, Key.END));
+   await articles[0].sendKeys(Key.chord(Key.CONTROL, Key.END)); // Briefly tested this and found that the `await` was also needed here
+   await t.context.session.switchTo().defaultContent();

    t.true(
      await checkFocus(t, ex.delayTimeSelector, 0),
      'Focus should move off the feed (onto element: ' +
        ex.delayTimeSelector +
        ') after sending keys CONTROL+END'
    );
  }
);
```

**Comment by ariellalgilmore** [comment]
Ahhhh thank you so much @howard-e!! Hopefully we can finally get this in now 😄

**Review by howard-e — APPROVED** [review]
Thanks for addressing the test feedback @ariellalgilmore! Following the update of the failing test, everything now seems good to me for the code and test related changes.

**Comment by a11ydoer** [comment]
@ariellalgilmore @mcking65 I am really sorry for last min notice. The example seem to fail the [color contrast ratio](https://webaim.org/resources/contrastchecker/?fcolor=777777&bcolor=FAFAFA) in normal text size. The current font color is #777 and backgroud color is # FAFAFA. The contrast ratio is only 4.29:1, not meeting 4.5:1

**Comment by a11ydoer** [comment]
High contrast support with forced-colors is functioning effectively.

**Review by a11ydoer — APPROVED** [review]
This is the approval for the corrected directory path.

**Comment by a11ydoer** [comment]
@ariellalgilmore one more thing, target size minimum rule, [WCAG SC 2.5.8 AA.](https://www.w3.org/WAI/WCAG22/Understanding/target-size-minimum.html)
The bookmark button height does not meet this requirment. Can you increase its height to 24px?

**Comment by ariellalgilmore** [comment]
Hi, I updated the PR today so that it the CTRL-END would go to the "terms-of-use" button, but the deploy preview is throwing an error for me: https://github.com/w3c/wai-aria-practices/actions/runs/8900404954/job/24449218445. Not sure if it has to do with this PR...

cc: @mcking65 @howard-e appreciate your eyes on this 👀

**Comment by mcking65** [comment]
@ariellalgilmore 

I merged #2997 and then merged main into this branch. all tests pass now.

However, ctrl+end is still going to the delay input for me.

**Comment by mcking65** [comment]
@ariellalgilmore  .... Oh, I now see that error. I thought if the preview failed, one of the checks would fail so when I saw the heading with all 25 checks passed, I assumed it was good. I now see that there is a preview build error at the end of the top comment.

@howard-e, please look into this!

**Comment by howard-e** [comment]
> @howard-e, please look into this!

@ariellalgilmore @mcking65 this should now be resolved! It was due to the generated PR falling out of sync in the build repository. This is already reported in https://github.com/w3c/wai-aria-practices/issues/219 and it can easily happen with older PRs. Maybe some priority needs to be applied to that in the near future.

> .... Oh, I now see that error. I thought if the preview failed, one of the checks would fail so when I saw the heading with all 25 checks passed, I assumed it was good. I now see that there is a preview build error at the end of the top comment.

Right, although most would check the build checks first so this does give a false impression that it was successful. I've created #3016 to track this.

**Review by mcking65 — APPROVED** [review]
Thank you @ariellalgilmore for all the work on this!

**Pull request merged**